components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/db/connector/CoordinatedTaskDBConnector.java

```diff
@@ -0,0 +1,362 @@
+                TaskQueryHelper.REMOVE_TASKS_OF_NODE)) {
+            preparedStatement.setString(1, nodeId);
+            preparedStatement.executeUpdate();
+        }
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Removed all the tasks of node [" + nodeId + "].");
+        }
+    }
+
+    /**
+     * Remove the task entry from the db.
+     *
+     * @param coordinatedTasks - List of tasks to be removed.
+     */
+    public void removeTasksFromDB(List<String> coordinatedTasks) throws SQLException {
+
+        if (coordinatedTasks.isEmpty()) {
+            return;
+        }
+        try (Connection connection = getConnection(); PreparedStatement preparedStatement = connection.prepareStatement(
+                TaskQueryHelper.DELETE_TASK)) {
+            for (String task : coordinatedTasks) {
+                preparedStatement.setString(1, task);
+                preparedStatement.addBatch();
+            }
+            preparedStatement.executeBatch();
+        }
+        debugLogs(coordinatedTasks, "Following list of tasks were deleted from data base.");
+    }
+
+    /**
+     * Helper method to populate coordinated tasks to the list.
+     *
+     * @param preparedStatement - Statement to be executed to retrieve the list of tasks.
+     * @param taskList          - The list to be populated.
+     * @throws SQLException - Exception.
+     */
+    private void populateCoordinatedTaskNames(PreparedStatement preparedStatement, List<String> taskList)
+            throws SQLException {
+        try (ResultSet resultSet = preparedStatement.executeQuery()) {
+            while (resultSet.next()) {
+                taskList.add(resultSet.getString(ConstantUtil.TASK_NAME));
+            }
+        }
+    }
+
+    /**
+     * Helper method to populate coordinated tasks to the list.
+     *
+     * @param preparedStatement - Statement to be executed to retrieve the list of tasks.
+     * @param taskList          - The list to be populated.
+     * @throws SQLException - Exception.
+     */
+    private void populateCoordinatedTaskList(PreparedStatement preparedStatement, List<CoordinatedTask> taskList)
+            throws SQLException {
+
+        try (ResultSet resultSet = preparedStatement.executeQuery()) {
+            while (resultSet.next()) {
+
+                taskList.add(new CoordinatedTask(resultSet.getString(ConstantUtil.TASK_NAME),
+                                                 resultSet.getString(ConstantUtil.DESTINED_NODE_ID),
+                                                 CoordinatedTask.States
+                                                         .valueOf(resultSet.getString(ConstantUtil.TASK_STATE))));
+            }
+        }
+    }
+
+    /**
+     * Retrieve all the tasks in the task db.
+     *
+     * @return - List of available tasks.
+     */
+    public List<String> getAllTasksInDB() throws SQLException {
```

Comment:
We are only returning taskNames, so shall we rename the method to imply that? Also we don;t need to say from DB, because this is the DBConnector.